Exiled.Events/Patches/Generic/PlayerHasHint.cs

```diff
@@ -17,50 +17,55 @@ namespace Exiled.Events.Patches.Generic
     /// Patches <see cref="HintDisplay.Show(Hint)"/>.
     /// </summary>
     [HarmonyPatch(typeof(HintDisplay), nameof(HintDisplay.Show))]
-    public static class PlayerHasHintPatch
+    public static class PlayerHasHint
     {
 #pragma warning disable SA1313 // Parameter names should begin with lower-case letter
 #pragma warning disable SA1600 // Elements should be documented
 #pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
         // Creating a list for coroutine check
-        private static List<Player> plyHaveCoroutine = new List<Player>();
+        private static HashSet<Player> playerHasHintCoroutines = new HashSet<Player>();
 
         public static void Postfix(HintDisplay __instance, Hint hint)
         {
             // Try to get the player, if it doesn't exist, just return
-            Player player = Player.Get(__instance.gameObject);
-            if (player == null)
+            if (!(Player.Get(__instance.gameObject) is Player player))
                 return;
 
-            // Set HasHint to true, and then delays to false for the duration of the hint
-            player.HasHint = true;
-
-            // Set HintDisplayTime to the duration of the last hint
-            // This prevents to set HasHint to false before the hint has finished
-            player.HintRemainingTime = hint.DurationScalar;
-
-            // If not contains, run coroutine
-            if (!plyHaveCoroutine.Contains(player))
+            // If contains, kill coroutine
+            if (playerHasHintCoroutines.Contains(player))
+            {
+                // Kill player coroutine
+                Timing.KillCoroutines(player.UserId + "HasHint");
+            }
+            else
             {
-                plyHaveCoroutine.Add(player);
-                Timing.RunCoroutine(HasHintToFalse(player));
+                // If it doesn't contains the player, add it
+                playerHasHintCoroutines.Add(player);
             }
+
+            // If it is false, then to true
+            if (!player.HasHint)
+                player.HasHint = true;
+
+            // I have to create a variable to be able to change the value of coroutine tag
+            CoroutineHandle coroutine = Timing.RunCoroutine(HasHintToFalse(player, hint.DurationScalar));
+
+            // Change the value of coroutine tag to the userid of the player
+            // Add HasHint part to ensure the tag is unique
+            coroutine.Tag = player.UserId + "HasHint";
```

Comment:
This is fine, but you could even use a `Dictionary<Player, CoroutineHandle>` to simplify the code and then, above, you could just use `playerHasHintCoroutines.TryGetValue(...)` to get the coroutine and kill it.

Reply:
Done, now using Dictionary, and i have added Player.ArtificialHealthDecay.